src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -74,16 +108,36 @@ public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
      * - "someDate is today" - если someDate - сегодня
      */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        LocalDate today = LocalDate.now();
+        if (someDate.isAfter(today)) {
+            return someDate + " is after " + today;
+        } else if (someDate.isBefore(today)) {
+            return someDate + " is before " + today;
+        } else {
+            return someDate + " is today";
+        }
     }
 
     /**
      * Даны две временные зоны.
      * Верните Optional часовой разницы между двумя временными зонами.
+     *
      * @return Optional positive Integer
      */
     public Optional<Integer> diffBetweenZones(String firstZone, String secondZone) {
-        return Optional.empty();
+        try {
+            ZoneId zone1 = ZoneId.of(firstZone);
+            ZoneId zone2 = ZoneId.of(secondZone);
+            LocalDateTime dateTime = LocalDateTime.now();
+            ZonedDateTime firstDateTime = ZonedDateTime.of(dateTime, zone1);
+            ZonedDateTime secondDateTime = ZonedDateTime.of(dateTime, zone2);
+            int first = firstDateTime.getOffset().getTotalSeconds() / 3600;
+            int second = secondDateTime.getOffset().getTotalSeconds() / 3600;
+            return Optional.of(Math.abs(first - second));
```

Comment:
```suggestion
            return Optional.of(Math.abs(firstDateTime.getHour() - secondDateTime.getHour()));
```

Reply:
In current moment, firstDateTime.getHour() and secondDateTime.getHour() return int 14 (actuall hour in real time). I tried to find method that can return hours, but as I understood, we can get only seconds or milliseconds and by dividing them get hours or what else we would need :)

Reply:
is it a problem to get 14?

Reply:
I'm sorry if i'm wrong, but firstDateTime.getHour() - secondDateTime.getHour() will always be 0, because in this two cases it will return the same number (14 now). And if we use get.Offset() it return seconds of timezone we set.
![example](https://user-images.githubusercontent.com/25364036/71582758-3e225880-2b14-11ea-83d8-8a8718796783.png)